sw/device/lib/testing/flash_ctrl_testutils.c

```diff
@@ -135,31 +162,43 @@ bool flash_ctrl_testutils_erase_page(
   return flash_ctrl_testutils_wait_transaction_end(flash_state);
 }
 
-bool flash_ctrl_testutils_write_page(
-    dif_flash_ctrl_state_t *flash_state, uint32_t byte_address,
-    uint32_t partition_id, const uint32_t *data,
-    dif_flash_ctrl_partition_type_t partition_type, uint32_t word_count) {
+bool flash_ctrl_testutils_write(dif_flash_ctrl_state_t *flash_state,
+                                uint32_t byte_address, uint32_t partition_id,
+                                const uint32_t *data,
+                                dif_flash_ctrl_partition_type_t partition_type,
+                                uint32_t word_count) {
+  // Cannot program partial flash words
+  if (byte_address & (FLASH_CTRL_PARAM_BYTES_PER_WORD - 1)) {
+    return false;
+  }
   dif_flash_ctrl_transaction_t transaction = {.byte_address = byte_address,
                                               .op = kDifFlashCtrlOpProgram,
                                               .partition_type = partition_type,
                                               .partition_id = partition_id,
                                               .word_count = 0x0};
+  // Assume the flash word width is greater than or equal to the bus word width.
+  // i.e. FLASH_CTRL_PARAM_BYTES_PER_WORD > sizeof(uint32_t)
```

Comment:
I've seem as a standard that when we have a documented code we put it between backtick.

Reply:
How about an actual static assertion instead? :)